packages/firestore/src/model/values.ts

```diff
@@ -559,8 +559,11 @@ export function isMapValue(
 export function deepClone(source: Value): Value {
   if (source.geoPointValue) {
     return { geoPointValue: { ...source.geoPointValue } };
-  } else if (source.timestampValue) {
-    return { timestampValue: { ...normalizeTimestamp(source.timestampValue) } };
+  } else if (
+    source.timestampValue &&
+    typeof source.timestampValue === 'object'
```

Comment:
I don't follow why we need the `typeof` check in the `else if` block. In what conditions does `timestampValue` not have type `object`? Do we need an separate condition to handle that, or does returning `{...source}` in the `else` statement work?

Reply:
The timestampValue can be a string, in which case `{...}` does not work. It would create an array of characters.